src/main/java/org/mifos/connector/channel/api/definition/GSMATransactionApi.java

```diff
@@ -10,5 +10,7 @@ public interface GSMATransactionApi {
     GsmaP2PResponseDto gsmatransaction(@RequestBody GsmaTransfer requestBody,
                                        @RequestHeader(value = "X-CorrelationID") String correlationId,
                                        @RequestHeader(value = "amsName") String amsName,
-                                       @RequestHeader(value = "accountHoldingInstitutionId") String accountHoldId) throws JsonProcessingException;
+                                       @RequestHeader(value = "accountHoldingInstitutionId") String accountHoldId,
```

Comment:
you cannot add these headers here. Remember we discussed that these headers should not appear in the channel connector code.  This code is generic not mpesa specific. we cannot add any header that is solution specific

about  reconciled and mpesaTxnId

Reply:
This is needed further in mpesa- connector for settlement flow.